File: 24-swap-nodes-in-pairs/24-swap-nodes-in-pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode* curr=NULL,*prev=NULL,*next=NULL;
        
        for(prev=head;prev && prev->next;prev=prev->next)
        {
            next=prev->next;
            prev->next=next->next;
            next->next=prev;
            
            if(curr!=NULL)
            {
                curr->next=next;
            }
            curr=prev;
            if(head==prev)
                head=next;
        }
        return head;
    }
};

File: 24-swap-nodes-in-pairs/24-swap-nodes-in-pairs.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {
    if head==nil || head.Next==nil{
        return head
    }
    res:=head.Next
    head.Next=swapPairs(head.Next.Next)
    res.Next=head
    return res
}